packages/compiler-cli/src/ngtsc/annotations/common/test/metadata_spec.ts

```diff
@@ -111,9 +111,24 @@ runInEachFileSystem(() => {
         `{ 'has-dashes-in-name': [{ type: Input }], noDashesInName: [{ type: Input }] })`,
       );
     });
+
+    it('should not emit metadata for ECMAScript private fields', () => {
+      const res = compileAndPrint(
+        `
+        import {Component, Input} from '@angular/core';
+
+        @Component('metadata') class Target {
```

Comment:
Using a directive here would make this legal Angular code:

```suggestion
        @Directive() class Target {
```